File: ios/FreelanceKar/Views/Screens/OrderScreen.swift
import SwiftUI

struct OrderScreen: View {
    @EnvironmentObject var store: AppStore
    private let seq: [BookingStatus] = [.awaiting, .in_progress, .delivered, .approved]

    var body: some View {
        let s = store.state
        if let b = Selectors.currentBooking(s) {
            let at = seq.firstIndex(of: b.status) ?? 0
            let complete = b.status == .approved

            ScreenShell {
                VStack(alignment: .leading, spacing: 0) {
                    HStack(spacing: 12) {
                        BackButton { store.go(.you) }
                        Text("Order \(b.id)").font(.fkScreenTitle)
                    }
                    .padding(.horizontal, 20).padding(.top, 16)

                    VStack(spacing: 0) {
                        HStack(spacing: 12) {
                            Stripe(cornerRadius: 13).frame(width: 44, height: 44)
                            VStack(alignment: .leading, spacing: 3) {
                                Text(b.name).font(.fkH4)
                                Text(b.service).font(.system(size: 11.5)).foregroundStyle(FK.text50)
                            }
                            Spacer()
                        }
                        HStack {
                            VStack(alignment: .leading, spacing: 0) {
                                Text(b.amount).font(.system(size: 18, weight: .semibold))
                                Text(complete ? "Paid to \(b.first) directly" : "Pay \(b.first) directly")
                                    .font(.system(size: 11))
                                    .foregroundStyle(complete ? FK.text45 : FK.lime)
                            }
                            Spacer()
                            Text(Selectors.orderStatusLabel[b.status] ?? "")
                                .font(.fkMonoSm)
                                .foregroundStyle(complete ? FK.text45 : FK.lime)
                        }
                        .padding(.top, 14)
                        .overlay(Rectangle().frame(height: 1).foregroundStyle(FK.line07), alignment: .top)
                        .padding(.top, 13)
                    }
                    .padding(16)
                    .background(FK.card)
                    .clipShape(RoundedRectangle(cornerRadius: 18, style: .continuous))
                    .overlay(RoundedRectangle(cornerRadius: 18).stroke(FK.line08, lineWidth: 1))
                    .padding(.horizontal, 20).padding(.top, 20)

                    VStack(alignment: .leading, spacing: 13) {
                        Text("Progress").font(.fkH3)
                        VStack(spacing: 0) {
                            ForEach(Array(Selectors.orderSteps.enumerated()), id: \.offset) { i, step in
                                let reached = i <= at
                                let isLast = i == Selectors.orderSteps.count - 1
                                HStack(alignment: .top, spacing: 13) {
                                    VStack(spacing: 0) {
                                        Circle()
                                            .fill(reached ? FK.lime : FK.bg)
                                            .frame(width: 11, height: 11)
                                            .overlay(Circle().stroke(reached ? FK.lime : FK.line18, lineWidth: 2))
                                        if !isLast {
                                            Rectangle().fill(i < at ? FK.lime : FK.line12).frame(width: 1).frame(minHeight: 26)
                                        }
                                    }
                                    .frame(width: 18)
                                    VStack(alignment: .leading, spacing: 2) {
                                        Text(step.0).font(.system(size: 13, weight: .semibold)).foregroundStyle(reached ? FK.text : FK.text40)
                                        Text(reached ? step.1 : "—").font(.system(size: 11)).foregroundStyle(FK.text45)
                                    }
                                    Spacer()
                                }
                                .padding(.bottom, isLast ? 0 : 16)
                            }
                        }
                    }
                    .padding(.horizontal, 20).padding(.top, 22)

                    if let review = b.review {
                        VStack(spacing: 0) {
                            HStack {
                                Text("Your review").font(.system(size: 13, weight: .semibold))
                                Spacer()
                                Text(String(repeating: "★", count: review.stars)).font(.system(size: 12, weight: .medium)).foregroundStyle(FK.amber)
                            }
                            Text(review.text).font(.fkBodySm).foregroundStyle(FK.text60).padding(.top, 9)
                                .frame(maxWidth: .infinity, alignment: .leading)
                        }
                        .padding(14)
                        .background(FK.card)
                        .clipShape(RoundedRectangle(cornerRadius: 15, style: .continuous))
                        .overlay(RoundedRectangle(cornerRadius: 15).stroke(FK.line08, lineWidth: 1))
                        .padding(.horizontal, 20).padding(.top, 4)
                    }

                    if let issue = b.issue {
                        VStack(alignment: .leading, spacing: 0) {
                            HStack {
                                Text("Issue · \(issue.ref)").font(.system(size: 13, weight: .semibold)).foregroundStyle(FK.alert)
                                Spacer()
                                Text(issue.status == .open ? "SUPPORT REVIEWING" : "SUPPORT REPLIED")
                                    .font(.fkMonoSm)
                                    .foregroundStyle(issue.status == .open ? FK.amber : FK.lime)
                            }
                            Text(issue.reason).font(.system(size: 11.5)).foregroundStyle(FK.text55).padding(.top, 5)
                            Text(issue.status == .open
                                 ? "Raised with the support team. Keep any files or messages from \(b.first.isEmpty ? "the freelancer" : b.first) — you’ll get a reply within 24 hours."
                                 : "Support has spoken to \(b.first.isEmpty ? "the freelancer" : b.first) and paused their listing while this is open. Add anything else in chat.")
                                .font(.fkBodySm).foregroundStyle(FK.text60).padding(.top, 9)
                        }
                        .padding(14)
                        .background(FK.alert08)
                        .clipShape(RoundedRectangle(cornerRadius: 15, style: .continuous))
                        .overlay(RoundedRectangle(cornerRadius: 15).stroke(FK.alertBd35, lineWidth: 1))
                        .padding(.horizontal, 20).padding(.top, 4)
                    }

                    VStack(spacing: 9) {
                        if b.status == .delivered {
                            PrimaryButton(title: "Mark complete & rate") { store.openReview() }
                        }
                        OutlineButton(title: "Message \(b.first)") { if !b.proId.isEmpty { store.openThread(b.proId) } }
                        if b.issue == nil {
                            Button { store.openIssue() } label: {
                                Text("Raise an issue").font(.system(size: 12, weight: .medium)).foregroundStyle(FK.text55)
                            }
                        }
                    }
                    .padding(.horizontal, 20).padding(.top, 18).padding(.bottom, 26)
                }
            }
        }
    }
}

File: android/app/src/main/java/com/freelancekar/app/ui/screens/OrderScreen.kt
package com.freelancekar.app.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.unit.dp
import com.freelancekar.app.data.*
import com.freelancekar.app.ui.AppViewModel
import com.freelancekar.app.ui.components.BackButton
import com.freelancekar.app.ui.components.OutlineButton
import com.freelancekar.app.ui.components.PrimaryButton
import com.freelancekar.app.ui.components.Stripe
import com.freelancekar.app.ui.theme.FkColor
import com.freelancekar.app.ui.theme.FkType

private val ORDER_SEQ = listOf(BookingStatus.AWAITING, BookingStatus.IN_PROGRESS, BookingStatus.DELIVERED, BookingStatus.APPROVED)

@Composable
fun OrderScreen(state: AppState, vm: AppViewModel) {
    val b = currentBooking(state) ?: return
    val at = ORDER_SEQ.indexOf(b.status)
    val complete = b.status == BookingStatus.APPROVED
    val issue = b.issue

    Column(Modifier.fillMaxSize().verticalScroll(rememberScrollState())) {
        Row(Modifier.fillMaxWidth().padding(horizontal = 20.dp).padding(top = 16.dp), verticalAlignment = Alignment.CenterVertically) {
            BackButton { vm.go(Screen.YOU) }
            Spacer(Modifier.width(4.dp))
            Text("Order ${b.id}", style = FkType.screenTitle)
        }

        Column(
            Modifier
                .padding(horizontal = 20.dp)
                .padding(top = 20.dp)
                .fillMaxWidth()
                .clip(RoundedCornerShape(18.dp))
                .background(FkColor.card)
                .border(1.dp, FkColor.line08, RoundedCornerShape(18.dp))
                .padding(16.dp),
        ) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Stripe(Modifier.size(44.dp).clip(RoundedCornerShape(13.dp)))
                Spacer(Modifier.width(12.dp))
                Column(Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(3.dp)) {
                    Text(b.name, style = FkType.h4)
                    Text(b.service, style = FkType.meta.copy(color = FkColor.text50))
                }
            }
            Row(
                Modifier.fillMaxWidth().padding(top = 14.dp).border(0.dp, FkColor.line07).padding(top = 13.dp),
                horizontalArrangement = Arrangement.SpaceBetween,
            ) {
                Column {
                    Text(b.amount, style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold, fontSize = androidx.compose.ui.unit.TextUnit(18f, androidx.compose.ui.unit.TextUnitType.Sp)))
                    Text(if (complete) "Paid to ${b.first} directly" else "Pay ${b.first} directly", style = FkType.meta.copy(color = if (complete) FkColor.text45 else FkColor.lime))
                }
                Text(ORDER_STATUS_LABEL[b.status] ?: "", style = FkType.monoSm.copy(color = if (complete) FkColor.text45 else FkColor.lime))
            }
        }

        Column(Modifier.padding(horizontal = 20.dp).padding(top = 22.dp)) {
            Text("Progress", style = FkType.h3)
            Spacer(Modifier.height(13.dp))
            ORDER_STEPS.forEachIndexed { i, (title, note) ->
                val reached = i <= at
                val last = i == ORDER_STEPS.size - 1
                Row(Modifier.fillMaxWidth()) {
                    Column(Modifier.width(18.dp), horizontalAlignment = Alignment.CenterHorizontally) {
                        Box(
                            Modifier
                                .size(11.dp)
                                .clip(CircleShape)
                                .background(if (reached) FkColor.lime else FkColor.bg)
                                .border(2.dp, if (reached) FkColor.lime else FkColor.line18, CircleShape),
                        )
                        if (!last) {
                            Box(Modifier.width(1.dp).height(26.dp).background(if (i < at) FkColor.lime else FkColor.line12))
                        }
                    }
                    Column(Modifier.weight(1f).padding(start = 13.dp, bottom = 16.dp), verticalArrangement = Arrangement.spacedBy(2.dp)) {
                        Text(title, style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold, color = if (reached) FkColor.text else FkColor.text40))
                        Text(if (reached) note else "—", style = FkType.meta.copy(color = FkColor.text45))
                    }
                }
            }
        }

        if (b.review != null) {
            Column(Modifier.padding(horizontal = 20.dp).padding(top = 4.dp)) {
                Column(
                    Modifier
                        .fillMaxWidth()
                        .clip(RoundedCornerShape(15.dp))
                        .background(FkColor.card)
                        .border(1.dp, FkColor.line08, RoundedCornerShape(15.dp))
                        .padding(14.dp),
                ) {
                    Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                        Text("Your review", style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold))
                        Text("★".repeat(b.review.stars), style = FkType.meta.copy(color = FkColor.amber))
                    }
                    Spacer(Modifier.height(9.dp))
                    Text(b.review.text, style = FkType.bodySm.copy(color = FkColor.text60))
                }
            }
        }

        if (issue != null) {
            Column(Modifier.padding(horizontal = 20.dp).padding(top = 4.dp)) {
                Column(
                    Modifier
                        .fillMaxWidth()
                        .clip(RoundedCornerShape(15.dp))
                        .background(FkColor.alert08)
                        .border(1.dp, FkColor.alertBd35, RoundedCornerShape(15.dp))
                        .padding(14.dp),
                ) {
                    Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                        Text("Issue · ${issue.ref}", style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold, color = FkColor.alert))
                        Text(if (issue.status == "open") "SUPPORT REVIEWING" else "SUPPORT REPLIED", style = FkType.monoSm.copy(color = if (issue.status == "open") FkColor.amber else FkColor.lime))
                    }
                    Spacer(Modifier.height(5.dp))
                    Text(issue.reason, style = FkType.meta.copy(color = FkColor.text55))
                    Spacer(Modifier.height(9.dp))
                    val firstOrDefault = b.first.ifEmpty { "the freelancer" }
                    Text(
                        if (issue.status == "open") "Raised with the support team. Keep any files or messages from $firstOrDefault — you'll get a reply within 24 hours."
                        else "Support has spoken to $firstOrDefault and paused their listing while this is open. Add anything else in chat.",
                        style = FkType.bodySm.copy(color = FkColor.text60),
                    )
                }
            }
        }

        Column(Modifier.padding(horizontal = 20.dp).padding(top = 18.dp, bottom = 26.dp), verticalArrangement = Arrangement.spacedBy(9.dp)) {
            if (b.status == BookingStatus.DELIVERED) {
                PrimaryButton({ vm.openReview() }, text = "Mark complete & rate")
            }
            OutlineButton({ b.proId.let { vm.openThreadPublic(it) } }, text = "Message ${b.first}")
            if (issue == null) {
                Text(
                    "Raise an issue",
                    style = FkType.meta.copy(color = FkColor.text55),
                    modifier = Modifier.fillMaxWidth().clickable { vm.openIssue() }.padding(6.dp),
                    textAlign = androidx.compose.ui.text.style.TextAlign.Center,
                )
            }
        }
    }
}
